### ISSUE TITLE
isGranted   is always true

### ISSUE THREAD

**Issue by yuxinLi** [tbruyelle/RxPermissions#149]
the code is like below: 

`new RxPermissions(getActivity())
                .request(CAMERA, RECORD_AUDIO)
                .subscribe(new Action1<Boolean>() {
                    @Override
                    public void call(Boolean isGranted) {
                        if (isGranted) {
                        
                        } else {
                        }
                    }
                });`

firstly , the  "isGranted"   is true , and then the phone show the permission dialog , I click refuse , 
but  next time , the  "isGranted"   is also true

**Comment by yuxinLi** [comment]
the android os version is  :  6.0.1

**Comment by tbruyelle** [comment]
> firstly , the "isGranted" is true , and then the phone show the permission dialog

No. The `Action1` is only invoked when user has answered all permission dialog requests.

**Comment by ChinaDragon01** [comment]
I have encountered the same problem, may I ask this question solved?

**Comment by epool** [comment]
@yuxinLi @ChinaDragon01 do you have some example project for reproducing it?

**Comment by neosdong** [comment]
I have encountered the same problem.

**Comment by gao746700783** [comment]
// when targetSdkVersion < Build.VERSION_CODES.M(23), calling like below:
        // ContextWrapper.checkSelfPermission or Context.checkSelfPermission ,
        // it will always return PERMISSION_GRANTED
i will give a pr of that
